examples/src/custom_kms_client_for_kms_provider.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Example showing how to customize the AWS KMS Client."""
+import aws_encryption_sdk
+from aws_encryption_sdk import CommitmentPolicy
+import boto3
+
+# Create a new class that extends the AWS KMS Provider you need to use
+class CustomKMSClientMasterKeyProvider(aws_encryption_sdk.StrictAwsKmsMasterKeyProvider):
+
+    # Override `add_regional_client` to use whatever configuration you need
+    def add_regional_client(self, region_name):
+        """Adds a regional client for the specified region if it does not already exist.
+        :param str region_name: AWS Region ID (ex: us-east-1)
+        """
+        if region_name not in self._regional_clients:
+            session = boto3.session.Session(botocore_session=self.config.botocore_session)
+            client = session.client(
+                'kms',
+                region_name=region_name,
+                config=self._user_agent_adding_config,
+                # Add additional custom client configuration here
+                connect_timeout=10
+            )
+            self._register_client(client, region_name)
+            self._regional_clients[region_name] = client
```

Comment:
Just a note: IMO by including this in the official examples, this implicitly blesses these private methods as part of the public API for this class. Not sure if that's an issue or not, depending on the plans for the future of the codebase. :)

Reply:
Yes, it may be better to refactor this example as:

```
main_mk = StrictAwsKmsMasterKeyProvider(providerConfig)
main_mk.add_master_key(MRKAwareKMSMasterKey(anotherConfig))
```

However this does not address discovery.